From 109cba51af0fc23f58f40d805abbcaceeba77f4a Mon Sep 17 00:00:00 2001
From: Nikhil Badola <nikhil.badola@freescale.com>
Date: Mon, 11 May 2015 17:21:27 +0530
Subject: [PATCH 722/741] usb: fsl: Correct endpoint name checking

commit 715d712c4b21dc6d0ee5df9610110a50c433f790 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Endpoint name being statically defined cannot be NULL
hence checking name's first two character with "ep" for
endpoint validity

Signed-off-by: Nikhil Badola <nikhil.badola@freescale.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/usb/gadget/udc/fsl_udc_core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/udc/fsl_udc_core.c b/drivers/usb/gadget/udc/fsl_udc_core.c
index a4eacebfdafe..6827874f6ae4 100644
--- a/drivers/usb/gadget/udc/fsl_udc_core.c
+++ b/drivers/usb/gadget/udc/fsl_udc_core.c
@@ -1701,7 +1701,7 @@ static void dtd_complete_irq(struct fsl_udc *udc)
 		curr_ep = get_ep_by_pipe(udc, i);
 
 		/* If the ep is configured */
-		if (!curr_ep->ep.name) {
+		if (strncmp(curr_ep->name, "ep", 2)) {
 			WARNING("Invalid EP?");
 			continue;
 		}
-- 
2.17.1

